internal/bootcamp/encode/encode.go

```diff
@@ -0,0 +1,47 @@
+package encode
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+)
+
+type EncodeResponse interface {
+	Encode(http.ResponseWriter, interface{}) error
+}
+
+type JsonResponseEncoder struct {
```

Comment:
when doing DI, we should hide implementation struct and only expose inteface and constructor